pkg/forklift-api/webhooks/validating-webhook/admitters/plan-admitter.go

```diff
@@ -116,6 +83,105 @@ func (admitter *PlanAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv
 			},
 		}
 	}
+
 	log.Info("Passed storage validation")
 	return util.ToAdmissionResponseAllow()
 }
+
+func (admitter *PlanAdmitter) validateVDDK() *admissionv1.AdmissionResponse {
+	if admitter.sourceProvider.Type() != api.VSphere {
+		log.Info("Provider type (non-VSphere) does not require VDDK, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	el9, el9Err := admitter.plan.VSphereUsesEl9VirtV2v()
+	if el9Err != nil {
+		log.Error(el9Err, "Could not analyze plan, failing")
+		return util.ToAdmissionResponseError(el9Err)
+	}
+	if el9 {
+		// VDDK image is optional when EL9 virt-v2v image is in use
+		log.Info("VDDK image is optional when EL9 virt-v2v image is in use, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	if _, found := admitter.sourceProvider.Spec.Settings["vddkInitImage"]; !found {
+		err := liberr.New("VDDK image is necessary for this type of migration")
+		log.Error(err, "VDDK image required for this type of migration")
+		return util.ToAdmissionResponseError(err)
+	}
+
+	return util.ToAdmissionResponseAllow()
+}
+
+func (admitter *PlanAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("Plan admitter was called")
+	raw := ar.Request.Object.Raw
+
+	err := json.Unmarshal(raw, &admitter.plan)
+	if err != nil {
+		return util.ToAdmissionResponseError(err)
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Error(err, "Couldn't get the cluster configuration")
+		return util.ToAdmissionResponseError(err)
+	}
+
+	err = api.SchemeBuilder.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Error(err, "Couldn't build the scheme")
+		return util.ToAdmissionResponseError(err)
+	}
+	err = apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Error(err, "Couldn't add forklift API to the scheme")
+		return util.ToAdmissionResponseError(err)
+	}
+
+	admitter.client, err = client.New(config, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		log.Error(err, "Couldn't create a cluster client")
+		return util.ToAdmissionResponseError(err)
+	}
+
+	err = admitter.client.Get(
+		context.TODO(),
+		client.ObjectKey{
+			Namespace: admitter.plan.Spec.Provider.Source.Namespace,
+			Name: admitter.plan.Spec.Provider.Source.Name,
+		},
+		&admitter.sourceProvider)
+	if err != nil {
+		log.Error(err, "Couldn't get the source provider")
+		return util.ToAdmissionResponseError(err)
```

Comment:
I reviewed the provider's admitter before this one - the comment on allowing the request when not all the data is available also applies here (it was incorrect in the original code, before your changes)